ProjFS.Mac/PrjFSKextTests/KauthHandlerTests.mm

```diff
@@ -58,4 +60,187 @@ - (void)testShouldIgnoreVnodeType {
     XCTAssertFalse(ShouldIgnoreVnodeType(static_cast<vtype>(1000), testVnode.get()));
 }
 
+- (void)testFileFlaggedInRoot {
+    bool fileFlaggedInRoot;
+    shared_ptr<mount> testMount = mount::Create();
+    shared_ptr<vnode> testVnode = vnode::Create(testMount, "/foo");
+    
+    testVnode->attrValues.va_flags = FileFlags_IsInVirtualizationRoot;
+    XCTAssertTrue(TryGetFileIsFlaggedAsInRoot(testVnode.get(), NULL, &fileFlaggedInRoot));
+    XCTAssertTrue(fileFlaggedInRoot);
+    
+    testVnode->attrValues.va_flags = FileFlags_IsEmpty;
+    XCTAssertTrue(TryGetFileIsFlaggedAsInRoot(testVnode.get(), NULL, &fileFlaggedInRoot));
+    XCTAssertFalse(fileFlaggedInRoot);
+    
+    testVnode->attrValues.va_flags = FileFlags_Invalid;
+    XCTAssertTrue(TryGetFileIsFlaggedAsInRoot(testVnode.get(), NULL, &fileFlaggedInRoot));
+    XCTAssertFalse(fileFlaggedInRoot);
+    
+    testVnode->attrValues.va_flags = 0x00000100;
+    XCTAssertTrue(TryGetFileIsFlaggedAsInRoot(testVnode.get(), NULL, &fileFlaggedInRoot));
+    XCTAssertFalse(fileFlaggedInRoot);
+
+    testVnode->errors.getattr = EBADF;
+    XCTAssertFalse(TryGetFileIsFlaggedAsInRoot(testVnode.get(), NULL, &fileFlaggedInRoot));
+}
+
+- (void)testShouldHandleVnodeOpEvent {
+    // In Parameters
+    shared_ptr<mount> testMount = mount::Create();
+    shared_ptr<vnode> testVnode = vnode::Create(testMount, "/foo");
+    testVnode->attrValues.va_flags = FileFlags_IsInVirtualizationRoot;
+    PerfTracer perfTracer;
+    vfs_context_t _Nonnull context = vfs_context_create(NULL);
+    kauth_action_t action = KAUTH_VNODE_READ_DATA;
+    
+    // Out Parameters
+    vtype vnodeType;
+    uint32_t vnodeFileFlags;
+    int pid;
+    char procname[MAXCOMLEN + 1] = "";
+    int kauthResult;
+    int kauthError;
+
+    
+    // Test Success Case
+    XCTAssertTrue(
+        ShouldHandleVnodeOpEvent(
+            &perfTracer,
+            context,
+            testVnode.get(),
+            action,
+            &vnodeType,
+            &vnodeFileFlags,
+            &pid,
+            procname,
+            &kauthResult,
+            &kauthError));
+    XCTAssertEqual(kauthResult, KAUTH_RESULT_DEFER);
+    
+    
+    // Test Invalid Action Bit
+    XCTAssertFalse(
+        ShouldHandleVnodeOpEvent(
+            &perfTracer,
+            context,
+            testVnode.get(),
+            KAUTH_VNODE_ACCESS,
```

Comment:
Also add a test for `KAUTH_VNODE_ACCESS` OR-ed with something we do otherwise care about?

e.g. `KAUTH_VNODE_ACCESS | KAUTH_VNODE_DELETE`